ospfd/ospf_vty.c

```diff
@@ -10415,6 +10440,14 @@ static int ospf_config_write_one(struct vty *vty, struct ospf *ospf)
 	if (ospf->passive_interface_default == OSPF_IF_PASSIVE)
 		vty_out(vty, " passive-interface default\n");
 
+	/* proactive-arp print. */
+	if (ospf->proactive_arp != OSPF_PROACTIVE_ARP_DEFAULT) {
+		if (ospf->proactive_arp)
```

Comment:
Another small detail: usually we only print the non default configuration, so I think this is a bit redundant (unless we change the default, but I guess that won't change long term).

Reply:
The default configuration isn't printed because of this condition
```
if (ospf->proactive_arp != OSPF_PROACTIVE_ARP_DEFAULT)
```
Or do I miss something?

Reply:
you are right, but this code can be simplified.

If `(ospf->proactive_arp != OSPF_PROACTIVE_ARP_DEFAULT)` then in theory (since a boolean can only be `true` or `false`, or in this case, `0` or `1`) the second condition will never trigger the first path `vty_out(vty, " proactive-arp\n");`.

Unless, of course, a bug happens and `ospf->proactive_arp` has a value different than `0` or `1` or someone changes the definition of `OSPF_PROACTIVE_ARP_DEFAULT`. But I wouldn't plan for bugs or changing constants.

This is just a nit, not a problem.